### PR TITLE
Squad Cognitive Crew Kai 7.2 Project Pull Request

### PR DESCRIPTION
**Pull request by PuneetPowar5**
Squad Cognitive Crew has finally completed their version of KAI 7.2 and it is ready to be checked before merging into the main branch. The PR request template has been attached to this request as well.
[PR request template.pdf](https://github.com/user-attachments/files/16751652/PR.request.template.pdf)

### CODE REVIEW

File: app/features/dynamo/doc_type_processors/ppt_processor.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    def generate_summary(self, data: pd.DataFrame) -> str:
+        summary = []
+        for _, row in data.iterrows():
+            row_summary = ", ".join([f"{col}: {row[col]}" for col in data.columns])
+            summary.append(row_summary)
+        return "\n".join(summary)
+
+    def generate_flashcards(self, summary: str, examples: str, format_instructions: str) -> List[Flashcard]:
+        prompt_template = (
+            "You are a flashcard generation assistant designed to help students analyze a document and return a list of flashcards. "
```
Comment (hash2004) [inline app/features/dynamo/doc_type_processors/ppt_processor.py:39]: You can use .txt file to store prompts and import it from there.

File: app/features/dynamo/doc_type_processors/test_ppt_processor.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# test_ppt_processor.py
+import pandas as pd
+from ppt_processor import PPTProcessor
+from langchain_google_vertexai import VertexAI  # Make sure this import is correct or mock it
+
+# Mock the VertexAI object if not available
+class MockVertexAI:
```
Comment (hash2004) [inline app/features/dynamo/doc_type_processors/test_ppt_processor.py:7]: Why are we mocking the VertexAI API if its not available with an empty class? Instead do robust error handling and edge case handling.

File: app/features/dynamo/tools.py [hunk 3]
```diff
@@ -28,7 +28,7 @@ def read_text_file(file_path):
         return file.read()
 
 # Summarize chain
-def summarize_transcript(youtube_url: str, max_video_length=600, verbose=False) -> str:
```
Comment (hash2004) [inline app/features/dynamo/tools.py:31]: max_video_length is a necessary part of the application. Please include it in the parameters as before.

File: app/features/dynamo/tools.py [hunk 4]
```diff
@@ -43,21 +43,44 @@ def summarize_transcript(youtube_url: str, max_video_length=600, verbose=False)
...
-        raise VideoTranscriptError(f"Video is {length} seconds long, please provide a video less than {max_video_length} seconds long", youtube_url)
-
     if verbose:
         logger.info(f"Found video with title: {title} and length: {length}")
         logger.info(f"Splitting documents into {len(split_docs)} chunks")
+        
+    if len(split_docs) > 10:
+        chain_type='map_reduce'
+    else:
+        chain_type='stuff'
```
Comment (hash2004) [inline app/features/dynamo/tools.py:81]: Haha, this is not the correct way to handle edge cases. Please revise the code.

**Review by hash2004 — CHANGES_REQUESTED** [review]
Please explain the changes you make by following the PR template. Thank you!

**Comment by AaronSosaRamos** [comment]
Closed due inactivity

**Pull request closed without merging**